### PR TITLE
refactor: Strict null checks

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- Enable strict nulls for some low-hanging fruits

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/metadata.ts [hunk 1]
```diff
@@ -189,22 +189,22 @@ function massageTimestamps(dep: ReleaseResult): void {
...
 
 export function addMetaData(
-  dep?: ReleaseResult,
-  datasource?: string,
-  lookupName?: string
+  dep: ReleaseResult,
+  datasource: string,
+  lookupName: string
 ): void {
-  if (!dep) {
```
Comment (zharinov) [inline lib/datasource/metadata.ts:196]: Unlikely to happen since the only call returns early for nullable `dep` and `datasource` values

**Review by rarkins — DISMISSED** [review]

File: lib/datasource/metadata.ts [hunk 2]
```diff
@@ -246,12 +246,18 @@ export function addMetaData(
   }
 
   // Clean up any empty urls
-  const urls = ['homepage', 'sourceUrl', 'changelogUrl', 'dependencyUrl'];
-  for (const url of urls) {
-    if (is.string(dep[url]) && validateUrl(dep[url].trim())) {
-      dep[url] = dep[url].trim();
+  const urlKeys: (keyof ReleaseResult)[] = [
```
Comment (viceice) [inline lib/datasource/metadata.ts:249]: Remove?

Comment (zharinov) [reply lib/datasource/metadata.ts]: Yup

**Review by viceice — DISMISSED** [review]

File: lib/datasource/metadata.ts (hunk 2 above)
Comment (zharinov) [reply lib/datasource/metadata.ts:249]: Not sure how to make it work without this type declaration, though

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**